cordapp/README.md

```diff
@@ -95,7 +107,337 @@ GqO4M1KlfimphQwCICiq00hDanT5W8bTLqE7GIGuplf/O8AABlpWrUg6uiUB
+to `await`. The flow will suspend and free up its thread, potentially for an extended period. The enclave need not reply immediately.
+It can return from processing the delivered mail without replying. When the enclave replies, the flow will be re-awakened, and the encrypted mail will be returned to the other side.
+
+```kotlin
+@Suspendable
+@Throws(FlowException::class)
+fun relayMessageToFromEnclave() {
+    // Other party sends us an encrypted mail.
+    val encryptedMail = session.receive(ByteArray::class.java).unwrap { it }
+    // Deliver and wait for the enclave to reply. The flow will suspend until the enclave chooses to deliver a mail
+    // to this flow, which might not be immediately.
+    val encryptedReply: ByteArray = flow.await(host.deliverAndPickUpMail(flow, encryptedMail))
+    // Send back to the other party the encrypted enclave's reply
+    session.send(encryptedReply)
+}
+```
+
+*Note:
+This sample code does not handle node restarts, although the Corda flow framework has built-in support for it.*
+
+### The Initiator Flow
+
+The initiator flow that starts a session with the responder party is as follows:
+
+1. Get the enclave attestation and validate it.
+2. Build and send a verifiable identity for the enclave to validate.
+3. Send the encrypted mail with the string to reverse.
+4. Read and decrypt the enclave's response. Remember that a verifiable identity is only sent to the enclave if the `anonymous`
+property is set to false.
+
+```java
+@InitiatingFlow
+@StartableByRPC
+public class ReverseFlow extends FlowLogic<String> {
+    private final Party receiver;
+    private final String message;
+    private final String constraint;
+    private final Boolean anonymous;
+
+    public ReverseFlow(Party receiver, String message, String constraint) {
+        this(receiver, message, constraint, false);
+    }
+
+    public ReverseFlow(Party receiver, String message, String constraint, Boolean anonymous) {
+        this.receiver = receiver;
+        this.message = message;
+        this.constraint = constraint;
+        this.anonymous = anonymous;
+    }
+
+    @Override
+    @Suspendable
+    public String call() throws FlowException {
+        EnclaveFlowInitiator session = EnclaveClientHelper.initiateFlow(this, receiver, constraint, anonymous);
+
+        byte[] response = session.sendAndReceive(message.getBytes(StandardCharsets.UTF_8));
+
+        return new String(response);
+    }
+}
+```
+
+After establishing a session with the hosting node, the flow waits for an [`EnclaveInstanceInfo`](api/-conclave/com.r3.conclave.common/-enclave-instance-info/index.html) to verify it
+against the constraint passed. If the enclave verifies successfully
+and doesn't throw an exception, the flow sends the initiator party identity to the enclave for
+[authentication](writing-cordapps.md#authenticating-senders-identity). The last step is applicable only if the party wishes to
```

Comment:
Link doesn't work. Maybe if we change it to README.md#authenticating-senders-identity? I am not sure how readme behaves with links though.

Reply:
Corrected by adding an anchor link in the document and referencing it.